app/lib/server/lib/interceptDirectReplyEmails.js

```diff
@@ -117,19 +136,32 @@ export class IMAPIntercepter {
 								email.headers.to = email.headers.to[0];
 								email.headers.date = email.headers.date[0];
 								email.headers.from = email.headers.from[0];
+								email.headers.subject = email.headers.subject[0];
+								if (email.headers.references && email.headers.references[0]) {
+									email.headers.references = email.headers.references[0];
+								}
 							}
 						});
 					});
 
 					// On fetched each message, pass it further
 					msg.once('end', Meteor.bindEnvironment(() => {
-						// delete message from inbox
-						if (this.delete) {
-							this.imap.seq.addFlags(seqno, 'Deleted', (err) => {
-								if (err) { console.log(`Mark deleted error: ${ err }`); }
-							});
+						// direct reply emails must have a message id (in its 'to' or 'subject' header) and a need a reference (in-reply-to)
+						if ((email.headers.subject.indexOf('reply:') > 0 || email.headers.to.indexOf('+') > 0) && email.headers.references) {
```

Comment:
only handle those emails having a reply messages referenced and a a reference (in-reply-to). Leave all other mails untouched by Rocket.Chat.